metaBuild.sh

```diff
@@ -1,65 +1,26 @@
-#!/bin/bash
-# TODO: Test config for OSX
-# TODO: Determine appropriate architecture (i.g. "args+=(arch=nocona)")
-
 DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-if [ -z "$3" ]; then
-  ARCH=x86-64
+ARG1="$1"
+shift
+CMAKE_ARGS=$@
+echo "CMAKE_ARGS: ${CMAKE_ARGS}"
+if [ "${ARG1}" = "debug" ]; then
+  CMAKE_ARGS+=" -DCMAKE_BUILD_TYPE=Debug "
+elif [ "${ARG1}" = "release" ]; then
+  CMAKE_ARGS+=" -DCMAKE_BUILD_TYPE=Release "
+elif [ "${ARG1}" = "clean" ]; then
+  (cd "${DIR}/build" && make clean)
+  exit 0
 else
-  ARCH=$3
+  echo "Missing or unrecognized metabuild argument \"$1\""
+  exit 1
 fi
-echo Architecture is set to $ARCH
-
-thirdparty_dir='#/..'
-
-if [ "$1" = "clean" ] || [ "$1" = "build" ]; then
-  declare -a args=()
-  SYSTEM_NAME=`uname -s`
-  args+=(sse=0)
-  args+=(arch=$ARCH)
-  if [ "$SYSTEM_NAME" = "Darwin" ]; then
-    # OSX specific config:
-    args+=(use_openmesh=1 openmesh_libpath=$thirdparty_dir/OpenMesh-2.0/build/Build/lib/OpenMesh openmesh_publiclibs='OpenMeshCore,OpenMeshTools' openmesh_include=$thirdparty_dir/OpenMesh-2.0/src)
-    echo "" # Can't have an empty if block
-  elif echo "$SYSTEM_NAME" | grep -q "MINGW64_NT"; then
-    # Windows specific config:
-    args+=(libs_extra=psapi)
-    args+=(use_openmesh=1 Werror=0 openmesh_libpath=$thirdparty_dir/OpenMesh-2.0/build/Build/lib openmesh_publiclibs='OpenMeshCore,OpenMeshTools' openmesh_include=$thirdparty_dir/OpenMesh-2.0/src)
-  elif echo "$SYSTEM_NAME" | grep -q "MSYS"; then
-    echo "ERROR: the MSYS shell is not supported. Please use the MinGW-w64 Win64 Shell instead."
-    exit 1
-  else
-    echo "ERROR: Unrecognized or unsupported platform: $SYSTEM_NAME!"
-    exit 1
-  fi
 
-  args+=(use_python=0 use_libjpeg=0 use_libpng=0 use_openexr=0 use_boost=0)
-  args+=(shared=1 install=0)
-  args+=(use_gmp=1 gmp_libpath=$thirdparty_dir/mpir/.libs/ gmp_include=$thirdparty_dir/mpir/)
+DIR_3RD_PARTY="${DIR}/../"
+CMAKE_ARGS+=" -DGMP_LIB_DIR=${DIR_3RD_PARTY}/mpir/.libs/ "
+CMAKE_ARGS+=" -DGMP_INCLUDE_DIR=${DIR_3RD_PARTY}/mpir/ "
+CMAKE_ARGS+=" -DCMAKE_VERBOSE_MAKEFILE=On "
+PKG_CONFIG_PATH="/System/Library/Frameworks/Python.framework/Versions/2.7/lib/pkgconfig:${PKG_CONFIG_PATH}"
```

Comment:
Make a note that this line is OSX only if you do not want to split this up into diff system builds quite yet.